repositories/strategies/duplicateStrategy.js

```diff
@@ -0,0 +1,230 @@
+const { omit, head, get, isNil } = require("lodash");
+const { getOrUpdateOrderForPageInsert } = require("./spacedOrderStrategy");
+
+const insertData = async (
+  trx,
+  tableName,
+  data,
+  callback = result => {
+    return result;
+  },
+  returning = "*",
+  position
+) => {
+  const returnedData = await trx
+    .table(tableName)
+    .insert(data)
+    .returning(returning)
+    .then(callback);
+
+  if (returnedData.order) {
+    const order = await getOrUpdateOrderForPageInsert(
+      trx,
+      returnedData.SectionId,
+      returnedData.id,
+      position
+    );
+
+    await trx
+      .table(tableName)
+      .update({ order: order })
+      .where({ id: returnedData.id });
+  }
+
+  return returnedData;
+};
+
+const selectData = async (trx, tableName, columns, where, callback) => {
+  return trx
+    .select(columns)
+    .from(tableName)
+    .where(where)
+    .then(callback);
```

Comment:
Unnecessary callback.  The promise is returned so the caller can add their own `.then()`